atom/browser/api/atom_api_app.cc

```diff
@@ -904,6 +904,38 @@ void App::GetFileIcon(const base::FilePath& path,
   }
 }
 
+v8::Local<v8::Value> App::GetAppMemoryInfo(v8::Isolate* isolate) {
+  AppIdProcessIterator processIterator;
+  auto processEntry = processIterator.NextProcessEntry();
+  mate::Dictionary result = mate::Dictionary::CreateEmpty(isolate);
+
+  while(processEntry != nullptr) {
+    int64_t pid = processEntry->pid();
+    auto process = base::Process::OpenWithExtraPrivileges(pid);
+
+    std::unique_ptr<base::ProcessMetrics> metrics(
+      base::ProcessMetrics::CreateProcessMetrics(process.Handle()));
+
+    mate::Dictionary dict = mate::Dictionary::CreateEmpty(isolate);
+    
+    dict.Set("workingSetSize",
+            static_cast<double>(metrics->GetWorkingSetSize() >> 10));
+    dict.Set("peakWorkingSetSize",
+            static_cast<double>(metrics->GetPeakWorkingSetSize() >> 10));
+
+    size_t private_bytes, shared_bytes;
+    if (metrics->GetMemoryBytes(&private_bytes, &shared_bytes)) {
+      dict.Set("privateBytes", static_cast<double>(private_bytes >> 10));
+      dict.Set("sharedBytes", static_cast<double>(shared_bytes >> 10));
+    }
+
+    result.Set(std::to_string(pid).c_str(), dict); 
```

Comment:
I don't think you need the `.c_str()` here, `result.Set` should handle `std::string` as the first argument.